pkg/apis/workspaces/v1alpha1/endpoint.go

```diff
@@ -6,12 +6,16 @@ package v1alpha1
 type EndpointProtocol string
 
 const (
-	// Endpoint will have either `http` or `https` traffic (according to the `Secure` field),
-	// typically on a TCP connection
+	// Endpoint will have `http` traffic, typically on a TCP connection.
+	// It will be automaticaly promoted to `https` when the `secure` field is set to `true`
 	HTTPEndpointProtocol  EndpointProtocol = "http"
-	// Endpoint will have either `ws` or `wss` traffic (according to the `Secure` field),
-	// typically on a TCP connection
+	// Endpoint will have `https` traffic, typically on a TCP connection
+	HTTPSEndpointProtocol  EndpointProtocol = "https"
+	// Endpoint will have `ws` traffic, typically on a TCP connection
+	// It will be automaticaly promoted to `wss` when the `secure` field is set to `true`
 	WSEndpointProtocol  EndpointProtocol = "ws"
+	// Endpoint will have `wss` traffic, typically on a TCP connection
+	WSSEndpointProtocol  EndpointProtocol = "wss"
```

Comment:
According to https://github.com/devfile/kubernetes-api/issues/33#issuecomment-629345779, this field should refer to transport protocol (i.e. TCP/UDP), and the URI protocol selection should be under `.attributes.scheme`.

Also: It's not clear to me why we need separate settings for `https/wss`, given that there is also a `secure` field. What happens if you specify `https` and `secure: false`? What if `scheme: http` and `secure: true`?

It's not clear to me why we have https/wss when there is also a `secure` attribute.

Reply:
> According to #33 (comment), this field should refer to transport protocol 

see this comment from @l0rd: https://github.com/devfile/kubernetes-api/pull/84#issuecomment-647348320

> It's not clear to me why we have https/wss when there is also a secure attribute.

see this comment https://github.com/devfile/kubernetes-api/pull/84#issuecomment-647472428 and the previous ones. Does it answer your question ?

> What if scheme: http and secure: true?

the `http` would be *auto-promoted* to `https` I assume, since TLS is a sort of logical pre-requisite for endpoint authentication. (mentioned in the schema doc)

Reply:
Yeah sorry I missed the earlier discussion.

Reply:
No problem :-)